aiohttp/web_response.py

```diff
@@ -609,27 +613,36 @@ def content_length(self, value):
         else:
             await super().write_eof()
 
-    async def _start(self, request):
+    async def _start(self, request, *args, **kwargs):
         if not self._chunked and hdrs.CONTENT_LENGTH not in self._headers:
             if not self._body_payload:
                 if self._body is not None:
                     self._headers[hdrs.CONTENT_LENGTH] = str(len(self._body))
                 else:
                     self._headers[hdrs.CONTENT_LENGTH] = '0'
 
-        return await super()._start(request)
+        return await super()._start(request, *args, **kwargs)
+
+    def _compress_body(self, zlib_mode):
+        compressobj = zlib.compressobj(wbits=zlib_mode)
+        self._compressed_body = \
+            compressobj.compress(self._body) + compressobj.flush()
 
-    def _do_start_compression(self, coding):
+    async def _do_start_compression(self, coding):
         if self._body_payload or self._chunked:
-            return super()._do_start_compression(coding)
+            return await super()._do_start_compression(coding)
         if coding != ContentCoding.identity:
             # Instead of using _payload_writer.enable_compression,
             # compress the whole body
             zlib_mode = (16 + zlib.MAX_WBITS
                          if coding.value == 'gzip' else -zlib.MAX_WBITS)
-            compressobj = zlib.compressobj(wbits=zlib_mode)
-            self._compressed_body = compressobj.compress(self._body) +\
-                compressobj.flush()
+            if self._zlib_thread_size is not None and \
+                    len(self._body) > self._zlib_thread_size:
+                await asyncio.get_event_loop().run_in_executor(
```

Comment:
Wouldn't it be better to have a thread pool in place?

Reply:
I think it would. If someone will set default executor as mutiprocessing one he will get very surprised by aiohttp behaviour. However, inplace thread pool will cause DoS situation - you need to limit overall threads by some sane number.

Reply:
Multiprocessing pool as default is deprecated by asyncio: it doesn't work well with API like `loop.getaddrinfo()` anyway.

Reply:
Perhaps ability to pass in thread pool? This can get complicated later if you want to support multiple pools

Reply:
Make sense. Feel free to update the PR.